src/test/transaction_tests.cpp

```diff
@@ -174,9 +174,29 @@ unsigned int FillFlags(unsigned int flags)
 // Assumes that mapFlagNames contains all script verify flags.
 std::set<unsigned int> ExcludeIndividualFlags(unsigned int flags)
 {
+    std::vector<unsigned int> set_flags;
+    const size_t n_flags = sizeof(unsigned int);
+    set_flags.reserve(n_flags);
+    for (size_t i = 0; i < n_flags; ++i) {
+        auto flag = flags & (1 << i);
+        if (flag) {
+            set_flags.push_back(flag);
+        }
+    }
+    const auto combos = 1 << set_flags.size();
+    // skip last all set flags
+    std::vector<unsigned int> all_flag_combos(combos-1);
+    for (size_t i = 0; i < combos-1; ++i){
+        for(size_t j = 0; j < set_flags.size(); ++j) {
+            if (i & (1<<j)) {
+                all_flag_combos[i] |= set_flags[j];
+            }
+        }
+    }
+
     std::set<unsigned int> flags_combos;
-    for (const auto& pair : mapFlagNames) {
-        const unsigned int flags_excluding_one = TrimFlags(flags & ~(pair.second));
+    for (const auto& excluded_flags : all_flag_combos) {
+        const unsigned int flags_excluding_one = TrimFlags(excluded_flags);
         if (flags != flags_excluding_one) {
```

Comment:
this check is actually probably superfluous with the iteration not including combos -1 (the all bits set).

Reply:
update: it's not, because of the TrimFlags

Reply:
Notice that `TrimFlags` only removes flags, so it is, indeed, superfluous:
`flags_excluding_one ⊆ excluded_flags ⊂ flags`

Having said that, there's no harm in leaving it here.